sdk/core/System.BinaryData/src/BinaryData.cs

```diff
@@ -0,0 +1,260 @@
+        /// <returns>A <see cref="BinaryData"/> instance.</returns>
+        public static BinaryData FromStream(Stream stream) =>
+            FromStreamAsync(stream, false).EnsureCompleted();
+
+        /// <summary>
+        /// Creates a <see cref="BinaryData"/> instance from the specified stream.
+        /// The passed in stream is not disposed by this method.
+        /// </summary>
+        /// <param name="stream">Stream containing the data.</param>
+        /// <param name="cancellationToken">An optional<see cref="CancellationToken"/> instance to signal
+        /// the request to cancel the operation.</param>
+        /// <returns>A <see cref="BinaryData"/> instance.</returns>
+        public static async Task<BinaryData> FromStreamAsync(
+            Stream stream,
+            CancellationToken cancellationToken = default) =>
+            await FromStreamAsync(stream, true, cancellationToken).ConfigureAwait(false);
+
+        private static async Task<BinaryData> FromStreamAsync(
+            Stream stream,
+            bool async,
+            CancellationToken cancellationToken = default)
+        {
+            if (stream == null)
+            {
+                throw new ArgumentNullException(nameof(stream));
+            }
+            if (stream.CanSeek && stream.Length > int.MaxValue)
+            {
+                throw new ArgumentOutOfRangeException(
+                    nameof(stream),
+                    "Stream length must be less than Int32.MaxValue");
+            }
+            using var memoryStream = new MemoryStream();
+            if (async)
+            {
+                await stream.CopyToAsync(memoryStream, CopyToBufferSize, cancellationToken).ConfigureAwait(false);
+            }
+            else
+            {
+                stream.CopyTo(memoryStream);
+            }
+            return new BinaryData(memoryStream.GetBuffer().AsMemory(0, (int) memoryStream.Position));
+        }
+
+
+        /// <summary>
+        /// Creates a <see cref="BinaryData"/> instance by serializing the passed in object using
+        /// the <see cref="JsonSerializer"/>.
+        /// </summary>
+        ///
+        /// <typeparam name="T">The type of the data.</typeparam>
+        /// <param name="jsonSerializable">The data to use.</param>
+        /// <param name="options">The <see cref="JsonSerializerOptions"/> to use when serializing to Json.</param>
+        ///
+        /// <returns>A <see cref="BinaryData"/> instance.</returns>
+        [Diagnostics.CodeAnalysis.SuppressMessage("Usage", "AZC0014:Avoid using banned types in public API", Justification = "<Pending>")]
+        public static BinaryData FromObjectAsJson<T>(
+            T jsonSerializable,
+            JsonSerializerOptions? options = default)
+        {
+            using var memoryStream = new MemoryStream();
+            var buffer = JsonSerializer.SerializeToUtf8Bytes(jsonSerializable, typeof(T), options);
```

Comment:
Using dotnet/runtime style rules, this `var` is not permitted.

```suggestion
            byte[] buffer = JsonSerializer.SerializeToUtf8Bytes(jsonSerializable, typeof(T), options);
```

After that change, it's obvious that this method can easily be rewritten without the MemoryStream.